### PR TITLE
Async calls to libvirt driver

### PR DESCRIPTION
**Pull request by stephanerobert**

### CODE REVIEW

File: virtualpdu/core.py [hunk 1]
```diff
@@ -42,13 +43,25 @@ def pdu_outlet_state_changed(self, pdu, outlet, state):
...
-            self.driver.power_off(device)
-            self.driver.power_on(device)
+            def switch_power():
+                self.driver.power_off(device)
+                self.driver.power_on(device)
+        else:
+            self.logger.error("Unknown power state: {}".format(state))
+            return
+
+        with ThreadPoolExecutor(max_workers=1) as executor:
```
Comment (fbouliane) [inline virtualpdu/core.py:63]: I would prefer seeing asyncio here As we are already [using asynccore](https://github.com/internap/virtualpdu/blob/master/virtualpdu/pdu/pysnmp_handler.py#L127) for the transport. A mixup of threadpool and asynccore doesn't seems a good Idea. However I would refactor this as an experiment and not related to this review.

File: virtualpdu/tests/integration/drivers/test_libvirt_driver.py [hunk 2]
```diff
@@ -49,3 +49,7 @@ def test_power_on_domain_not_found(self):
     def test_power_off_domain_not_found(self):
         self.assertRaises(drivers.DeviceNotFound,
                           self.driver.power_off, 'i-dont-exist')
+
+    def test_get_power_domain_not_found_raises(self):
```
Comment (fbouliane) [inline virtualpdu/tests/integration/drivers/test_libvirt_driver.py:53]: nice catch

**Review by fbouliane — APPROVED** [review]
👍

File: virtualpdu/core.py (hunk 1 above)
Comment (stephanerobert) [reply virtualpdu/core.py:63]: Interesting point. I honestly went with the family when I implemented ThreadPoolExecutor

**Review by JoProvost — APPROVED** [review]

**Pull request merged**